### PR TITLE
[Keyboard] Fixes to make snes_macropad play nicer with qmk configurator

### PR DESCRIPTION
**Pull request by JBarberU**
Some minor tweaks to make snes_macropad work better with QMK Configurator

## Description

Contains the following changes:

- changes to info.json to layout the lower 12 keys in the shape of a snes controller, to make mapping in configurator less confusing.
- changes the default implementation for get_layer_name_user so that it will print the layer number instead of "Unknown"

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/snes_macropad/snes_macropad.c [hunk 1]
```diff
@@ -9,6 +9,8 @@ char     key_name = ' ';
 uint16_t last_keycode;
 uint8_t  last_row;
 uint8_t  last_col;
+char     layer_str[3] = ""; // 2 digits + null terminator
+_Static_assert(MAX_LAYER < 100, "layer_str is too small");
```
Comment (zvecr) [inline keyboards/snes_macropad/snes_macropad.c:13]: QMK does not support more than 32 layers.
```suggestion
```

File: keyboards/snes_macropad/snes_macropad.c [hunk 2]
```diff
@@ -58,7 +60,8 @@ static void oled_render_keylog(void) {
 }
 
 __attribute__((weak)) const char * get_layer_name_user(int layer) {
-    return "Unknown";
+    snprintf(layer_str, sizeof(layer_str), "%d", layer);
+    return layer_str;
```
Comment (zvecr) [inline keyboards/snes_macropad/snes_macropad.c:64]: Assuming the current use case of  `get_layer_name_user`,
```suggestion
    return get_u8_str(layer_str, ' ');
```

File: keyboards/snes_macropad/snes_macropad.c [hunk 3]
```diff
@@ -9,6 +9,8 @@ char     key_name = ' ';
 uint16_t last_keycode;
 uint8_t  last_row;
 uint8_t  last_col;
+char     layer_str[3] = ""; // 2 digits + null terminator
```
Comment (zvecr) [inline keyboards/snes_macropad/snes_macropad.c:12]: Not required if using `get_u8_str`
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by JBarberU** [comment]
Thanks for the review, really appreciate it! I implemented your suggestion which certainly made the code neater.

**Review by zvecr — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**